### PR TITLE
Direct lemmas and inference ids for sygus extension

### PR DESCRIPTION
**Pull request by ajreynol**
This adds inference ID for the datatypes sygus solver, and changes its style to send lemmas instead of passing them to the caller.

### CODE REVIEW

**Review by nafur — CHANGES_REQUESTED** [review]
A bunch of style issues, but nothing major.

File: src/theory/datatypes/sygus_extension.cpp [hunk 1]
```diff
@@ -1644,18 +1660,11 @@ void SygusExtension::check( std::vector< Node >& lemmas ) {
...
-      }
     }
   }
 }
 
-bool SygusExtension::checkValue(Node n,
-                                  Node vn,
-                                  int ind,
-                                  std::vector<Node>& lemmas)
+bool SygusExtension::checkValue(Node n, Node vn, int ind)
```
[the remark this answers is not in the document]
Comment (ajreynol) [reply src/theory/datatypes/sygus_extension.cpp:1667]: n definitely needs to be Node, I'll change vn to TNode.

File: src/theory/datatypes/sygus_extension.cpp [hunk 2]
```diff
@@ -1255,22 +1273,23 @@ void SygusExtension::addSymBreakLemmasFor( TypeNode tn, Node t, unsigned d, Node
...
+          d_im.lemma(slem, InferenceId::DATATYPES_SYGUS_SYM_BREAK);
         }
       }
     }
   }
   Trace("sygus-sb-debug2") << "...finished." << std::endl;
 }
 
-void SygusExtension::preRegisterTerm( TNode n, std::vector< Node >& lemmas  ) {
+void SygusExtension::preRegisterTerm(TNode n)
```
[the remark this answers is not in the document]
Comment (ajreynol) [reply src/theory/datatypes/sygus_extension.cpp:1284]: It is TNode here.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**